fetch/fetch.go

```diff
@@ -629,6 +578,46 @@ func (f *Fetch) GetHash(hash types.Hash32, h datastore.Hint, validateHash bool)
 	return resChan
 }
 
+// GetLayerData get layer data from peers.
+func (f *Fetch) GetLayerData(ctx context.Context, lid types.LayerID, okCB func([]byte, p2p.Peer, int), errCB func(error, p2p.Peer, int)) error {
+	remotePeers := f.host.GetPeers()
+	numPeers := len(remotePeers)
+	if numPeers == 0 {
+		return errNoPeers
+	}
+
+	for _, p := range remotePeers {
+		peer := p
```

Comment:
why do we need this copy?

Reply:
this is the classic error of passing an iteration variable to function executed in a separate goroutine.
it's the same thing that we have `tc := tc` in the test setup.

this remind me that i haven't added tests for this. now added. PTAL @evt

Reply:
I've missed goroutine which is probably run in `f.lyrSrv.Request` :)